commcare_connect/opportunity/models.py

```diff
@@ -633,7 +634,37 @@ class VisitReviewStatus(models.TextChoices):
     disagree = "disagree", gettext("Disagree")
 
 
+class UserVisitQuerySet(models.QuerySet):
+    def with_any_flags(self, flags):
+        from commcare_connect.utils.flags import Flags
+
+        # flags should be a subset of Flags
+        allowed_flags = {flag.value for flag in Flags}
+        flags = list(set(flags) & allowed_flags)
+
+        if not flags:
+            return self
+
+        conditions = " || ".join([f"@[0] == $f{i}" for i in range(len(flags))])
+
+        params = []
+        for i, f in enumerate(flags):
+            params.extend([f"f{i}", f])
+
+        sql = f"""
+            jsonb_path_exists(
```

Comment:
I have tested these queries for performance on superset prod (replication db), and found them to be fast enough.